ext/json11/json11.cpp

```diff
@@ -93,18 +93,10 @@ static void dump(const string &value, string &out) {
             out += "\\r";
         } else if (ch == '\t') {
             out += "\\t";
-        } else if (static_cast<uint8_t>(ch) <= 0x1f) {
+        } else if (static_cast<uint8_t>(ch) <= 0x1f || static_cast<uint8_t>(ch) >= 0x7f) {
```

Comment:
you should not emit `\u0000` that is not allowed in json strings. You may not emit anything that looks like surrogate, if you aren't using it as one,  and also any unicode > 9999 has to be encoded with surrogate pair. See https://en.wikipedia.org/wiki/UTF-16#Code_points_from_U+010000_to_U+10FFFF

Reply:
After some discussion on irc iit turned out that \x0000 is not an issue.

The other issue is that not everything is encoded in a form that can be translated back. 
For the moment we feel that that is not a big issue. The most important part is to produce valid UTF8 to not confuse clients.